### ISSUE TITLE
Option to get the maps generated in the dim test

### ISSUE THREAD

**Issue by shwilks** [acorg/Racmacs#15]
Option to get the maps generated in the dim test

**Comment by skepner** [comment]
Sam, have you meant adding this issue to acmacs.r?

**Comment by shwilks** [comment]
Hi Eu, yes, sorry if the notifications are annoying, Mathilde has requested some new Racmacs features and I was just keeping a track of the ones I think I would need to ask you for input regarding additional acmacs.r features on.  This is one of them.

Sam

**Comment by skepner** [comment]
Well, you have assigned this to me but I am not a Racmacs contributor.

**Comment by shwilks** [comment]
Fair point, probably better to reference you with an @skepner

**Comment by skepner** [comment]
Do you mean saving all the projections generated during the test?

**Comment by shwilks** [comment]
Yes, I think this is what @mathrichard and @Adindakok mean.  I guess it would also be useful to know which antigens and sera were the test ones for each run.  They also have a question about how many optimization runs are done in the dimension test when creating projections for the training subset.  And also how many cross-validation repeats are performed for each dimension.

**Comment by skepner** [comment]
"I guess it would also be useful to know which antigens and sera were the test ones for each run." - sorry, I don't understand what this means.

Saving all maps is theoretically possible (but I don't quite understand what people are going to do with so many maps). Please provide more detailed description of how you expect accessing results. I can imaging two variants:
1. All maps are saved in some directory on the disk. Saving many huge charts takes a lot of time and usually takes more time than performing the test. I.e. if saving is requested, test is 3x time slower (I think).
2. Maps are kept in memory and Racmacs has access to them. May lead to R crash due to running out of memory or slowing down computer due to swapping. Perhaps I am wrong.

**Comment by shwilks** [comment]
I will leave @mathrichard and @Adindakok to answer the questions about what they plan to do with the data, but I think variant 1 sounds like a smart way to do it.

I guess if we have the maps saved in a directory we can then simply work out which were the test antigens and sera by seeing which titers have been set to * in each map table?

The slow down would be acceptable if people want to run it with this option enabled, already your implementation of dimension testing runs pretty fast for me.

**Comment by mathrichard** [comment]
Hi Eu
The reason we asked is because at some point (and I think that you discussed these issues with Stefan), it appears that they were some maps with better R2 in the dim test than that of our actual map. However, I went back to past email conversations, and there was a confusion about what correlation coefficient was being calculated. Stefan referred to the R2 and you to the Pearson coefficient.

Another question about the R2 calculated in the dim test: does the regression take the threshold values (<10) into account?

Thanks
Cheers
M

**Comment by skepner** [comment]
@mathrichard 
Hi Mathilde,
Threshold titers do not contribute to the regression.
Eu

**Comment by skepner** [comment]
implemented in acmacs.r v4.0
